### PR TITLE
[feature](nereids)Convert the expression from nereids to stale expr.

### PR DESCRIPTION
**Pull request by Kikyou1997**
# Proposed changes

 Add ExpressionConverter.java to convert the expression from nereids to stale expression.

## Problem Summary:

We need the stale expr when build fragment to execute query, so add this class to do this job.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No/)
3. Has document been added or modified: (No)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/PhysicalPlanTranslator.java [hunk 1]
```diff
@@ -63,7 +64,6 @@
 /**
  * Used to translate to physical plan generated by new optimizer to the plan fragments.
  */
-@SuppressWarnings("rawtypes")
 public class PhysicalPlanTranslator extends PlanOperatorVisitor<PlanFragment, PlanContext> {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/PhysicalPlanTranslator.java:67]: we need to call PlanNode#convertToVectoriezd at the end of translate

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExpressionConverter.java [hunk 2]
```diff
@@ -17,18 +17,169 @@
...
+        List<Expr> paramList = new ArrayList<>();
+        for (Expression expr : function.getFnParams().getExpression()) {
+            paramList.add(visit(expr, context));
+        }
+        return new FunctionCallExpr(function.getFnName().toString(), paramList);
+    }
+
+    @Override
+    public Expr visitBetweenPredicate(BetweenPredicate betweenPredicate, PlanContext context) {
+        return new org.apache.doris.analysis.BetweenPredicate(
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExpressionConverter.java:146]: @EmmyMiao87 could be execute between predicate expr? if be cannot do that, we should throw exception if there have a BetweenPredicate

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExpressionConverter.java [hunk 3]
```diff
@@ -17,18 +17,169 @@
...
+                throw new RuntimeException(String.format("Unknown node type: %s", nodeType.name()));
+        }
+        return new org.apache.doris.analysis.CompoundPredicate(staleOp,
+                visit(compoundPredicate.child(0), context),
+                visit(compoundPredicate.child(1), context));
+    }
+
+    @Override
+    public Expr visitArithmetic(Arithmetic arithmetic, PlanContext context) {
+        return new ArithmeticExpr(arithmetic.getArithOperator().getStaleOp(),
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExpressionConverter.java:180]: currently, Arithmetic has both unary expr and binary expr

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Arithmetic.java [hunk 4]
```diff
@@ -37,27 +39,42 @@ enum OperatorPosition {
...
+                Arithmetic.OperatorPosition.BINARY_INFIX, TExprOpcode.BITOR, Operator.BITOR),
+        BITXOR("^", "bitxor",
+                Arithmetic.OperatorPosition.BINARY_INFIX, TExprOpcode.BITXOR, Operator.BITXOR),
+        BITNOT("~", "bitnot",
+                Arithmetic.OperatorPosition.UNARY_PREFIX, TExprOpcode.BITNOT, Operator.BITNOT);
 
         private final String description;
         private final String name;
         private final Arithmetic.OperatorPosition pos;
         private final TExprOpcode opcode;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Arithmetic.java:65]: i think this is useless, remove it

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExpressionConverter.java [hunk 5]
```diff
@@ -17,18 +17,169 @@
...
+        } else if (dataType instanceof StringType) {
+            return new StringLiteral((String) literal.getValue());
+        }
+        throw new RuntimeException(String.format("Unsupported data type: %s", dataType.toString()));
+    }
+
+    @Override
+    public Expr visitFunctionCall(FunctionCall function, PlanContext context) {
+        List<Expr> paramList = new ArrayList<>();
+        for (Expression expr : function.getFnParams().getExpression()) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExpressionConverter.java:138]: getExpression rename it to getExpressions or getExpressionList

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExpressionConverter.java [hunk 6]
```diff
@@ -17,18 +17,169 @@
...
+            paramList.add(visit(expr, context));
+        }
+        return new FunctionCallExpr(function.getFnName().toString(), paramList);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExpressionConverter.java:141]: add a todo to translator distinct or count(*)

File: fe/fe-core/src/main/java/org/apache/doris/nereids/types/DataType.java [hunk 7]
```diff
@@ -61,4 +61,8 @@ public static DataType convertFromCatalogDataType(Type catalogType) {
     }
 
     public abstract Type toCatalogDataType();
+
+    public String typeName() {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/types/DataType.java:65]: unuse?

**Review by morrySnow — DISMISSED** [review]
LGTM

**Review by EmmyMiao87 — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Pull request merged**